e2e/app/component/dialog.ts

```diff
@@ -1,36 +1,32 @@
 import {ElementHandle, Page} from 'puppeteer';
+import Container from 'app/element/container';
+import * as xpathDefaults from 'app/element/xpath-defaults';
 
 export enum ButtonLabel {
   Confirm = 'Confirm',
   KeepEditing = 'Keep Editing',
 }
 
-const SELECTOR = {
-  dialogRoot: '.ReactModal__Content[role="dialog"]',
-  button: '[role="button"]'
-}
 
-export default class Dialog {
-  private dialogElement: ElementHandle;
+export default class Dialog extends Container {
 
-  constructor(private readonly page: Page) {
+  private dialogElement: ElementHandle;
 
+  constructor(private readonly puppeteerPage: Page, selector: {xpath?: string, testId?: string}) {
+    super(selector);
   }
 
   async getContent(): Promise<string> {
     await this.findDialog();
```

Comment:
Good points. Hopefully, there won't be need to add many new methods in future.
I could call `getDialog` in `constructor` but I really don't want to do that because I don't control when user start to create an instance of Dialog object. If instance is created before Dialog is visible, then it will error.

Reply:
+1: if you need to initialize something, then it shouldn't be named `findDialog()` maybe `initDialog()` if that's the case. But this is an anti-pattern anyway, and violates RAII.

If the pattern is that we need to do some kind of action to bind this class to a real, visible dialog, then that should be very explicit. But that's semantic coupling though.